### PR TITLE
ISSUE #2388: Fix show detail section

### LINKED ISSUE #2388: Show details/show less feature turned into a header for analysts
### Current Behavior

Currently analysts don't see the show details button as a button that shows/hides info. Instead they see the following
image
 This applies to both the organization and address sections, not the status history section

### Expected Behavior

The show details button should return to the old functionality, where it is a clickable button.

ETA: For both Domain and Domain Request when logged in as an Analyst

### Steps to Reproduce

1. log in as an analyst
2. go to a domain request
3. scroll down or search for "show details" and see it is not a button everywhere

### Environment

stable and sandboxes

### Additional Context

bug most likely resulted from  #2337 as it was released and then this bug occurred and it changed permissions on Domain requests/ domain info.

I would suggest to stop using permission checking to hide this field and instead we cover this with the org_model feature flag.

### Issue Links

_No response_

### PR DESCRIPTION
**Pull request by therealslimhsiehdy**
## Ticket

Resolves #2388 

## Changes

* When an Analyst comes in, the "show details" is nested under the `Type of organization` and `Organization name and mailing address` section

## Context for reviewers

Previously when an Analyst logged in, the show details seem to be separated/un-nested for those two sections

## Setup

1. Log in as an Analyst
2. Go to `/admin` in my sandbox
3. Click on `Domain requests` and click on any domain request you'd like
4. Check that `Show details` is nested under `Type of organization` and `Organization name and mailing address` and clickable

#### Satisfied acceptance criteria and met development standards

- [ ] Met the acceptance criteria, or will meet them in a subsequent PR
- [ ] Created/modified automated tests
- [ ] Added at least 2 developers as PR reviewers (only 1 will need to approve)
- [ ] Messaged on Slack or in standup to notify the team that a PR is ready for review
- [ ] Changes to “how we do things” are documented in READMEs and or onboarding guide
- [ ] If any model was updated to modify/add/delete columns, makemigrations was ran and the associated migrations file has been commited.

#### Ensured code standards are met (Original Developer)

- [ ] All new functions and methods are commented using plain language
- [ ] Did dependency updates in Pipfile also get changed in requirements.txt?
- [ ] Interactions with external systems are wrapped in try/except
- [ ] Error handling exists for unusual or missing values

#### Validated user-facing changes (if applicable)

- [ ] New pages have been added to .pa11yci file so that they will be tested with our automated accessibility testing
- [ ] Checked keyboard navigability
- [ ] Tested general usability, landmarks, page header structure, and links with a screen reader (such as Voiceover or ANDI)
- [ ] Add at least 1 designer as PR reviewer

#### Reviewed, tested, and left feedback about the changes

- [ ] Pulled this branch locally and tested it
- [ ] Reviewed this code and left comments
- [ ] Checked that all code is adequately covered by tests
- [ ] Made it clear which comments need to be addressed before this work is merged
- [ ] If any model was updated to modify/add/delete columns, makemigrations was ran and the associated migrations file has been commited.

#### Ensured code standards are met (Code reviewer)

- [ ] All new functions and methods are commented using plain language
- [ ] Interactions with external systems are wrapped in try/except
- [ ] Error handling exists for unusual or missing values
- [ ] (Rarely needed) Did dependency updates in Pipfile also get changed in requirements.txt?

#### Validated user-facing changes as a developer

- [ ] New pages have been added to .pa11yci file so that they will be tested with our automated accessibility testing
- [ ] Checked keyboard navigability
- [ ] Meets all designs and user flows provided by design/product
- [ ] Tested general usability, landmarks, page header structure, and links with a screen reader (such as Voiceover or ANDI)

- [ ] Tested with multiple browsers, the suggestion is to use ones that the developer didn't (check off which ones were used)
  - [ ] Chrome
  - [ ] Microsoft Edge
  - [ ] FireFox
  - [ ] Safari

- [ ] (Rarely needed) Tested as both an analyst and applicant user

**Note:** Multiple code reviewers can share the checklists above, a second reviewers should not make a duplicate checklist

#### Verified that the changes match the design intention

- [ ] Checked that the design translated visually
- [ ] Checked behavior
- [ ] Checked different states (empty, one, some, error)
- [ ] Checked for landmarks, page heading structure, and links
- [ ] Tried to break the intended flow

#### Validated user-facing changes as a designer

- [ ] Checked keyboard navigability
- [ ] Tested general usability, landmarks, page header structure, and links with a screen reader (such as Voiceover or ANDI)

- [ ] Tested with multiple browsers (check off which ones were used)
  - [ ] Chrome
  - [ ] Microsoft Edge
  - [ ] FireFox
  - [ ] Safari

- [ ] (Rarely needed) Tested as both an analyst and applicant user

### CODE REVIEW

File: src/registrar/admin.py [hunk 1]
```diff
@@ -1698,7 +1698,7 @@ def get_fieldsets(self, request, obj=None):
             for name, data in fieldsets:
                 fields = data.get("fields", [])
                 fields = tuple(field for field in fields if field not in self.superuser_only_fields)
-                modified_fieldsets.append((name, {"fields": fields}))
+                modified_fieldsets.append((name, {**data, "fields": fields}))
```
Comment (erinysong) [inline src/registrar/admin.py:1701]: Just out of curiosity, what does adding **data as a param do when appending to our updated fieldsets, and what makes this case different from the other show/hide detail collapses?

**Review by erinysong — APPROVED** [review]
LGTM! Confirmed show/details collapse successfully displays on fieldsets Type of Org, Org Name and Mailing Address) and Domain Request for DomainInfo, DomainRequest, and Domain models

File: src/registrar/admin.py (hunk 1 above)
Comment (therealslimhsiehdy) [reply src/registrar/admin.py:1701]: This is a SUPER good question, and honestly it was like 23842834 blog posts discovery later that I found this and this was also best explanation:

* Without `**data`: It's like taking an item (the fields) out of a box and putting it in a new box without the original settings (like collapsibility), so we lose the settings we have on the fields

* With `**data`: It's like taking the items out of the box carefully, changing just the fields, and putting everything back in the same box and then this way all the original settings (like collapsibility) stay intact

**Review by CocoByte — APPROVED** [review]
LGTM! (tested and verified)

Nice work -- this is not a straightforward fix

**Pull request merged**